senatus/src/index.ts

```diff
@@ -20,34 +23,49 @@ export default {
   async bootstrap({ strapi }: { strapi: Strapi }) {
     console.log("Senatus is bootstrapping");
     console.log(strapi.config.get("server.app.rabbitUrl"));
+    const prefetchCount = strapi.config.get<number>("server.app.prefetchResults");
     const { 
       subscribe,
-      publish,
+      disconnect,
     } = await connect(strapi.config.get("server.app.rabbitUrl"));
+
+    disconnectRabbit = disconnect;
+
     console.log("Connected to RabbitMQ!");
+    
     await subscribe("results", async (msg) => {
-      const parsed = JSON.parse(msg);
-      const submission = await strapi.entityService.findOne('api::submission.submission', parsed.metadata.submissionId, {
-        populate: ['user', 'problem', 'language'],
-      });
-
-      for (const result of parsed.result) {
-        console.log("Updating execution", result.metadata.executionId);
-        const existing = await strapi.entityService.findOne('api::execution.execution', result.metadata.executionId, {
-          populate: ['testCase'],
-        });
-        const execution = await strapi.entityService.update('api::execution.execution', result.metadata.executionId, {
-          data: {
-            stdout: result.stdout,
-            stderr: result.stderr,
-            executionTime: result.time,
-            processed: true,
-            passed: existing.testCase.output === result.stdout,
-            processedAt: new Date(),
-            publishedAt: new Date(),
-          }
+      try {
+        const parsed = JSON.parse(msg);
+        const submission = await strapi.entityService.findOne('api::submission.submission', parsed.metadata.submissionId, {
+          populate: ['user', 'problem', 'language'],
         });
+
+        for (const result of parsed.result) {
+          console.log("Updating execution", result.metadata.executionId);
+          const existing = await strapi.entityService.findOne('api::execution.execution', result.metadata.executionId, {
+            populate: ['testCase'],
+          });
+          const execution = await strapi.entityService.update('api::execution.execution', result.metadata.executionId, {
+            data: {
+              stdout: result.stdout,
+              stderr: result.stderr,
+              executionTime: result.time,
+              processed: true,
+              passed: existing.testCase.output === result.stdout,
```

Comment:
I think there was a small issue with stdout adding extra newline at the end, maybe if it's a small thing you could remove it now